### PR TITLE
[Feat] Add Radix Sort

### PR DESCRIPTION
**Pull request by rytheo**
A radix sort implementation for issue #3.

I considered using the existing `counting_sort`, but I can't tell it what radix and digit to use when sorting.

### CODE REVIEW

File: src/sorting/radix_sort.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/// When `n` and `b` are roughly the same maginitude, this algorithm runs in linear time.
+///
+/// Space complexity is `O(n + b)`.
+pub fn radix_sort(arr: &mut [u64]) {
+    let max: usize = match arr.iter().max() {
+        Some(&x) => x as usize,
+        None => return,
+    };
+    // Make radix a power of 2 close to arr.len() for optimal runtime
+    let radix = {
```
Comment (xfix) [inline src/sorting/radix_sort.rs:14]:
```suggestion
    let radix = arr.len().next_power_of_two();
```

File: src/sorting/radix_sort.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/// When `n` and `b` are roughly the same maginitude, this algorithm runs in linear time.
+///
+/// Space complexity is `O(n + b)`.
+pub fn radix_sort(arr: &mut [u64]) {
+    let max: usize = match arr.iter().max() {
+        Some(&x) => x as usize,
+        None => return,
+    };
+    // Make radix a power of 2 close to arr.len() for optimal runtime
+    let radix = arr.len().next_power_of_two();
```
Comment (AnshulMalik) [inline src/sorting/radix_sort.rs:14]: How does this help for optimal runtime?

File: src/sorting/radix_sort.rs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    while place <= max {
+        let digit_of = |x| x as usize / place % radix;
+        // Count digit occurrences
+        let mut counter = vec![0; radix];
+        for &x in arr.iter() {
+            counter[digit_of(x)] += 1;
+        }
+        // Compute last index of each digit
+        for i in 1..radix {
+            counter[i] += counter[i - 1];
```
Comment (AnshulMalik) [inline src/sorting/radix_sort.rs:26]: how does this for loop adds into the complexity?

File: src/sorting/radix_sort.rs (hunk 2 above)
Comment (rytheo) [reply src/sorting/radix_sort.rs:14]: It's a simple way to choose a radix of the same magnitude as `n`, which is necessary for the algorithm to run in linear time.

File: src/sorting/radix_sort.rs (hunk 3 above)
Comment (rytheo) [reply src/sorting/radix_sort.rs:26]: This loop has `O(b)` runtime where `b` is the radix. Therefore, this loop contributes the `b` of the `(n + b)` factor in the overall complexity.

Comment (AnshulMalik) [reply src/sorting/radix_sort.rs:26]: I see, this loop iterates `radix` times, not base of radix times, or am i missing something?

Comment (AnshulMalik) [reply src/sorting/radix_sort.rs:26]: okay got it.

**Review by AnshulMalik — APPROVED** [review]

**Pull request merged**